src/protocollogin.cpp

```diff
@@ -179,6 +192,20 @@ void ProtocolLogin::onRecvFirstMessage(NetworkMessage& msg)
 	}
 
 	std::string password = msg.getString();
+	if (password.empty()) {
+		disconnectClient("Invalid password.", version);
+		return;
+	}
+
+	// read authenticator token and stay logged in flag from last 128 bytes
+	msg.setBufferPosition(msg.getLength() - 128);
```

Comment:
Client always send it? i have get the impression that you can't login without token anymore.